### PR TITLE
device input generation in join bench

### PR DESCRIPTION
**Pull request by karthikeyann**
Use device functions to move input generation to device in join benchmark.

Splitting PR https://github.com/rapidsai/cudf/pull/10109 for review

### CODE REVIEW

File: cpp/benchmarks/join/join_common.hpp [hunk 1]
```diff
@@ -35,6 +40,18 @@
 
 #include "generate_input_tables.cuh"
 
+struct null75_generator {
```
Comment (vyasr) [inline cpp/benchmarks/join/join_common.hpp:43]: Would something like this be useful in other benchmarks? Should we make it parametrizable and move it to some common file?

**Review by vyasr — APPROVED** [review]
One minor suggestion for you to consider.

File: cpp/benchmarks/join/join_common.hpp (hunk 1 above)
Comment (karthikeyann) [reply cpp/benchmarks/join/join_common.hpp:43]: Yes. I have a parametrizable code for other benchmarks in a common file. That will be in a different PR, together with more changes.

**Review by PointKernel — APPROVED** [review]
LGTM :+1:

**Comment by karthikeyann** [comment]
rerun tests

**Comment by karthikeyann** [comment]
@gpucibot merge

**Pull request merged**